scripts/speculative_modes_bench.py

```
@@ -0,0 +1,468 @@
+# ----------------------------------------------------------------------------
+# Driver
+# ----------------------------------------------------------------------------
+async def run_cell(call_fn, approach: str, model_label: str) -> Dict[str, Any]:
+    ttfts, totals = [], []
+    valid_n, correct_n, capt_n = 0, 0, 0
+    valid_total, correct_total, capt_total = 0, 0, 0
+    for (ut, exp_line, exp_fn, exp_rating) in CASES:
+        for _ in range(N_ITERS):
+            try:
+                ttft, total, text, tcs = await call_fn(approach, ut)
+            except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)